src/Subscription.ts

```diff
@@ -38,12 +46,26 @@ export class Subscription {
       const len = _subscriptions.length;
 
       while (++index < len) {
-        const subscription = _subscriptions[index];
-        if (isObject(subscription)) {
-          subscription.unsubscribe();
+        const sub = _subscriptions[index];
+        if (isObject(sub)) {
+          let trial = tryCatch(sub.unsubscribe).call(sub);
```

Comment:
@blesh is there a reason this is `let trial` instead of `const trial`? same for above